File: AlgorithmicToolbox/FibonacciSumLastDigit.cpp
// Problem Description: Compute the last digit of F0 +F1 +···+Fn.

#include <iostream>
#include <vector>
#include <numeric>
using namespace std;

vector<int> f = {0, 1};

int fibsum(long long n)
{
    return (((n / f.size()) * accumulate(f.begin(), f.end(), 0) + accumulate(f.begin(), f.begin() + (n % f.size() + 1), 0)) % 10);
}

int main()
{
    int curr;
    long long n;
    for (int i = 2; i <= 100; i++)
    {
        curr = (f[i - 1] + f[i - 2]) % 10;
        if (curr == 0 && f[i - 1] == 1)
        {
            break;
        }
        f.push_back(curr);
    }
    cin >> n;
    cout << fibsum(n) << endl;
    return 0;
}

File: AlgorithmicToolbox/FibonacciSumLastDigit.py
# Problem Description: Compute the last digit of F0+F1+···+Fn.

import doctest

f = [0, 1]


def fibsum(n):
    """Returns sum of last digits of n terms of Fibonacci series.

    Examples:
    >>> fibsum(3)
    4
    >>> fibsum(100)
    5
    """
    return (
        # (n // len(f)) * sum(f) +
        # As sum(f)%10==0, remove the above line.
        sum(f[:n % len(f) + 1])) % 10


if __name__ == "__main__":
    for i in range(2, 101):
        curr = (f[i-1] + f[i-2]) % 10
        if(curr == 0 and f[i-1] == 1):
            break
        f.append(curr)
    doctest.testmod()
    n = int(input())
    print(fibsum(n))
